diff --git a/src/engine.h b/src/engine.h
index f5d09d4e..01a69863 100644
--- a/src/engine.h
+++ b/src/engine.h
@@ -92,6 +92,13 @@ class Engine {
     void                                 load_network(const std::string& file);
     void save_network(std::pair<std::optional<std::string>, std::string> file);
 
+    void load_big_network(std::istream& is) {
+        network.modify_and_replicate(
+          [this, &is](Eval::NNUE::Network& network_) { network_.load(is); });
+        threads.clear();
+        threads.ensure_network_replicated();
+    }
+
     // utility functions
 
     void trace_eval() const;
diff --git a/src/main.cpp b/src/main.cpp
index c386c418..ea9bf686 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -28,6 +28,8 @@
 
 using namespace Stockfish;
 
+UCIEngine* uci_global;
+
 #ifdef UNIVERSAL_BINARY
 namespace Stockfish {
 
@@ -43,11 +45,14 @@ int main(int argc, char* argv[]) {
     Attacks::init();
     Position::init();
 
-    auto uci = std::make_unique<UCIEngine>(argc, argv);
+    uci_global = new UCIEngine(0, nullptr);
 
-    Tune::init(uci->engine_options());
+    Tune::init(uci_global->engine_options());
 
-    uci->loop();
+    uci_global->loop();
+
+    delete uci_global;
+    uci_global = nullptr;
 
     return 0;
 }
diff --git a/src/misc.cpp b/src/misc.cpp
index d144a26f..14bc016c 100644
--- a/src/misc.cpp
+++ b/src/misc.cpp
@@ -508,6 +508,7 @@ bool is_whitespace(std::string_view s) {
 }
 
 std::string CommandLine::get_binary_directory(std::string argv0) {
+    return "";
     std::string pathSeparator;
 
 #ifdef _WIN32
@@ -542,6 +543,7 @@ std::string CommandLine::get_binary_directory(std::string argv0) {
 }
 
 std::string CommandLine::get_working_directory() {
+    return "";
     std::string workingDirectory = "";
     char        buff[40000];
     char*       cwd = GETCWD(buff, 40000);
diff --git a/src/nnue/network.cpp b/src/nnue/network.cpp
index 382f4f44..94268d4f 100644
--- a/src/nnue/network.cpp
+++ b/src/nnue/network.cpp
@@ -160,6 +160,9 @@ NetworkOutput Network::evaluate(const Position&    pos,
 
 void Network::verify(std::string                                  evalfilePath,
                      const std::function<void(std::string_view)>& f) const {
+
+    return; // Yep
+
     if (evalfilePath.empty())
         evalfilePath = evalFile.defaultName;
 
diff --git a/src/nnue/network.h b/src/nnue/network.h
index 5d7ba066..b914846f 100644
--- a/src/nnue/network.h
+++ b/src/nnue/network.h
@@ -74,14 +74,15 @@ class Network {
                                  AccumulatorStack&  accumulatorStack,
                                  AccumulatorCaches& cache) const;
 
+    std::optional<std::string> load(std::istream&);
+
    private:
     void load_user_net(const std::string&, const std::string&);
     void load_internal();
 
     void initialize();
 
-    bool                       save(std::ostream&, const std::string&, const std::string&) const;
-    std::optional<std::string> load(std::istream&);
+    bool save(std::ostream&, const std::string&, const std::string&) const;
 
     bool read_header(std::istream&, std::uint32_t*, std::string*) const;
     bool write_header(std::ostream&, std::uint32_t, const std::string&) const;
diff --git a/src/timeman.h b/src/timeman.h
index 08e8da10..917d27ec 100644
--- a/src/timeman.h
+++ b/src/timeman.h
@@ -48,7 +48,7 @@ class TimeManagement {
     TimePoint elapsed(FUNC nodes) const {
         return useNodesTime ? TimePoint(nodes()) : elapsed_time();
     }
-    TimePoint elapsed_time() const { return now() - startTime; };
+    TimePoint elapsed_time() const { return std::max(now() - startTime, TimePoint(1)); };
 
     void clear();
     void advance_nodes_time(std::int64_t nodes);
diff --git a/src/uci.cpp b/src/uci.cpp
index 77d62e91..f9290c2f 100644
--- a/src/uci.cpp
+++ b/src/uci.cpp
@@ -40,6 +40,8 @@
 #include "types.h"
 #include "ucioption.h"
 
+std::string js_getline();  // src/dualnet-glue.cpp
+
 namespace Stockfish {
 
 constexpr auto BenchmarkCommand = "speedtest";
@@ -65,8 +67,8 @@ void UCIEngine::print_info_string(std::string_view str) {
 }
 
 UCIEngine::UCIEngine(int argc, char** argv) :
-    engine(argv[0]),
-    cli(argc, argv) {
+    engine(),
+    cli(0, nullptr) {
 
     engine.get_options().add_info_listener([](const std::optional<std::string>& str) {
         if (str.has_value())
@@ -97,6 +99,8 @@ void UCIEngine::loop() {
             && !getline(std::cin, cmd))  // Wait for an input or an end-of-file (EOF) indication
             cmd = "quit";
 
+        cmd = js_getline();
+
         std::istringstream is(cmd);
 
         token.clear();  // Avoid a stale if getline() returns nothing or a blank line
@@ -172,7 +176,7 @@ void UCIEngine::loop() {
             sync_cout << "Unknown command: '" << cmd << "'. Type help for more information."
                       << sync_endl;
 
-    } while (token != "quit" && cli.argc == 1);  // The command-line arguments are one-shot
+    } while (token != "quit");  // The command-line arguments are one-shot
 }
 
 Search::LimitsType UCIEngine::parse_limits(std::istream& is) {
diff --git a/src/uci.h b/src/uci.h
index cfebc8af..4a16aa40 100644
--- a/src/uci.h
+++ b/src/uci.h
@@ -56,8 +56,9 @@ class UCIEngine {
 
     auto& engine_options() { return engine.get_options(); }
 
-   private:
     Engine      engine;
+
+   private:
     CommandLine cli;
 
     static void print_info_string(std::string_view str);
